tests/Fixer/Phpdoc/PhpdocTagNoNamedArgumentsFixerTest.php

```diff
@@ -0,0 +1,309 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Phpdoc;
+
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+use PhpCsFixer\WhitespacesFixerConfig;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Phpdoc\PhpdocTagNoNamedArgumentsFixer
+ *
+ * @extends AbstractFixerTestCase<\PhpCsFixer\Fixer\Phpdoc\PhpdocTagNoNamedArgumentsFixer>
+ *
+ * @phpstan-import-type _AutogeneratedInputConfiguration from \PhpCsFixer\Fixer\Phpdoc\PhpdocTagNoNamedArgumentsFixer
+ */
+final class PhpdocTagNoNamedArgumentsFixerTest extends AbstractFixerTestCase
+{
+    public function testInvalidConfiguration(): void
+    {
+        $invalidDirectory = __DIR__.'/directory-404';
+
+        $this->expectException(InvalidFixerConfigurationException::class);
+        $this->expectExceptionMessage(\sprintf('[%s] The directory "%s" does not exists.', $this->fixer->getName(), $invalidDirectory));
+
+        $this->fixer->configure(['directory' => $invalidDirectory]);
+    }
+
+    /**
+     * @param _AutogeneratedInputConfiguration $configuration
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input = null, array $configuration = [], ?WhitespacesFixerConfig $whitespacesFixerConfig = null): void
+    {
+        $this->fixer->configure($configuration);
+        if (null !== $whitespacesFixerConfig) {
+            $this->fixer->setWhitespacesConfig($whitespacesFixerConfig);
+        }
+        $this->doTest($expected, $input, new \SplFileInfo(__DIR__.'/../..'));
+    }
+
+    /**
+     * @return iterable<string, array{0: string, 1?: null|string, 2?: _AutogeneratedInputConfiguration}>
+     */
+    public static function provideFixCases(): iterable
+    {
+        yield 'do not add for anonymous class' => [
+            <<<'PHP'
+                <?php
+                new class () {};
+                PHP,
+        ];
+
+        yield 'do not add for different directory' => [
+            <<<'PHP'
+                <?php
+                class Foo {}
+                PHP,
+            null,
+            ['directory' => __DIR__.'/../../../src/Fixer'],
+        ];
+
+        yield 'do not add for internal class' => [
```

Comment:
why not ?

Reply:
Why repeat the information?

`@no-named-arguments` - no BC promise for the parameter names
`@internal` - no BC for anything

Reply:
What is your point?

Reply:
i suggest to add `@no-named-arguments` also for `@internal` classes

Reply:
Always or as a new option in a fixer?

Reply:
for this repo, i would suggest always.
for usage outside of main repo, i'm open to have it configurable (but not sure if needed - please decide)

Reply:
added config option